holidays/countries/saint_helena_ascension_and_tristan_da_cunha.py

```diff
@@ -0,0 +1,177 @@
+        * [2025](https://web.archive.org/web/20241005191212/https://www.sainthelena.gov.sh/wp-content/uploads/2024/08/Public-Notice-Public-Holidays-2025.pdf)
+    """
+
+    country = "SH"
+    default_language = "en_GB"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # Earliest year of holidays with an accessible online record.
+    start_year = 2015
+    subdivisions = (
+        "AC",  # Ascension.
+        "HL",  # Saint Helena.
+        "TA",  # Tristan da Cunha.
+    )
+    subdivisions_aliases = {
+        "Ascension": "AC",
+        "Saint Helena": "HL",
+        "Tristan da Cunha": "TA",
+    }
+    supported_languages = ("en_GB", "en_US")
+    supported_categories = (GOVERNMENT, PUBLIC)
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, cls=SaintHelenaAscensionAndTristanDaCunhaStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        if self._year <= 2022:
+            # Queen's Birthday.
+            name = tr("Queen's Birthday")
+            if self._year <= 2019:
+                self._add_holiday_3rd_mon_of_jun(name)
+            else:
+                self._add_holiday_2nd_mon_of_jun(name)
+        else:
+            kings_birthday_dates = {
+                2023: (JUN, 19),
+                2024: (NOV, 15),
+                2025: (NOV, 14),
+            }
+            # His Majesty King Charles III Birthday.
+            name = tr("His Majesty King Charles III Birthday")
+            if dt := kings_birthday_dates.get(self._year):
+                self._add_holiday(name, dt)
+
+        # Christmas Day.
+        self._add_observed(
+            self._add_christmas_day(tr("Christmas Day")), rule=SAT_SUN_TO_NEXT_MON_TUE
+        )
+
+        # Boxing Day.
+        self._add_observed(
+            self._add_christmas_day_two(tr("Boxing Day")), rule=SAT_SUN_TO_NEXT_MON_TUE
+        )
+
+    def _populate_subdiv_ac_public_holidays(self):
+        # Ascension Day.
+        self._add_ascension_thursday(tr("Ascension Day"))
+
+    def _populate_subdiv_hl_public_holidays(self):
+        # Saint Helena Day.
+        self._add_observed(self._add_holiday_may_21(tr("Saint Helena Day")))
+
+    def _populate_subdiv_ta_public_holidays(self):
+        # Ascension Day.
+        self._add_ascension_thursday(tr("Ascension Day"))
+
+        ratting_day_dates = {
```

Comment:
Source is https://www.tristandc.com/newsratting.php? Needs to be added to references.